### PR TITLE
Create Post routes for sign up/in/out

### PR DESCRIPTION
**Pull request by tracyyh**
## Description

Create a new “auth” package

Create sign up, sign in, sign out tRPC procedures

Write tests for routes

## Motivation and Context

Closes [story](https://trello.com/c/ThynemUk/14-post-routes-for-sign-up-sign-in-sign-out)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] CI/CD changes (changing github actions or deployment scripts)
- [x] New package(s)
- [ ] New dependencies
- [ ] Global style changes
- [ ] Project configuration

## Checklist:

- [x] I verified my changes work in the local environment
- [x] I verified my changes work in the preview environment
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## How has this been tested?
made tests for each new route
Ran bun test

### CODE REVIEW

File: .github/workflows/bun-tests.yml
Comment (owens1127) [inline .github/workflows/bun-tests.yml:1]: Changes in this file should fix the db issue for tests

File: apps/db/prisma/schema.prisma [hunk 1]
```diff
@@ -28,7 +28,7 @@ model Session {
   id Int @id @default(autoincrement())
```
Comment (owens1127) [inline apps/db/prisma/schema.prisma:28]: I did some research, [https://thecopenhagenbook.com/sessions](https://thecopenhagenbook.com/sessions), let's change this schema a bit.

We should delete the `id` field and then rename `token` to `id` based on how we are using it.

File: packages/trpc/src/procedures/auth.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import { z } from "zod";
+import { baseProcedureBuilder } from "../internal/init";
+const bcrypt = require('bcryptjs');
```
Comment (owens1127) [inline packages/trpc/src/procedures/auth.ts:3]: We cannot use 'require' statements in ESM modules (import syntax). Also, bcrypt will need to be added as a dependency to this package

File: packages/trpc/src/procedures/auth.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+import { comparePassword, hashPassword } from "@good-dog/auth";
+
+import { baseProcedureBuilder } from "../internal/init";
+
+// TODO: refactor to use cookies
+export const signUpProcedure = baseProcedureBuilder
+  .input(
+    z.object({
+      email: z.string(),
```
Comment (ispapa) [inline packages/trpc/src/procedures/auth.ts:11]: You can validate the format of the email string with `z.email()` as well

**Review by ispapa — COMMENTED** [review]
One minor comment otherwise looks good!

**Review by owens1127 — APPROVED** [review]

**Pull request merged**